archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterGenericInterfacesTest.java

```diff
@@ -0,0 +1,545 @@
+
+        assertThatType(getGenericInterface(child, InterfaceWithThreeSimpleTypeParameters.class)).as("generic interface")
+                .hasActualTypeArguments(String.class, Serializable.class, File.class);
+    }
+
+    @Test
+    public void imports_generic_superclass_and_multiple_generic_interfaces_in_combination() {
+        @SuppressWarnings("unused")
+        class BaseClass<X> {
+        }
+        @SuppressWarnings("unused")
+        class Child<T> extends BaseClass<File>
+                implements InterfaceWithSimpleTypeParameter<Path>, InterfaceWithTwoSimpleTypeParameters<T, String> {
+        }
+
+        JavaClass child = new ClassFileImporter()
+                .importClasses(Child.class,
+                        InterfaceWithSimpleTypeParameter.class, InterfaceWithTwoSimpleTypeParameters.class, InterfaceWithThreeSimpleTypeParameters.class,
+                        Path.class, List.class, Map.class, String.class, Serializable.class, File.class)
+                .get(Child.class);
+
+        assertThatType(child.getSuperclass().get())
+                .hasErasure(BaseClass.class)
+                .hasActualTypeArguments(File.class);
+
+        assertThatType(getGenericInterface(child, InterfaceWithSimpleTypeParameter.class)).as("generic interface")
+                .hasActualTypeArguments(Path.class);
+
+        assertThatType(getGenericInterface(child, InterfaceWithTwoSimpleTypeParameters.class)).as("generic interface")
+                .hasActualTypeArguments(typeVariable("T"), concreteClass(String.class));
+    }
+
+    @Test
+    public void imports_nested_generic_interfaces() {
+        @SuppressWarnings("unused")
+        class Child<T> implements
+                SomeDeeplyNestedInterface<File, SomeNestedInterface<Path, Path>> {
+        }
+
+        JavaType genericInterface = getOnlyElement(
+                new ClassFileImporter()
+                        .importClasses(Child.class, SomeDeeplyNestedInterface.class, SomeNestedInterface.class, File.class, Path.class)
+                        .get(Child.class).getInterfaces());
+
+        assertThatType(genericInterface).as("generic interface")
+                .hasErasure(SomeDeeplyNestedInterface.class)
+                .hasActualTypeArguments(
+                        concreteClass(File.class),
+                        parameterizedType(SomeNestedInterface.class).withTypeArguments(Path.class, Path.class));
+    }
+
+    private JavaType getGenericInterface(JavaClass javaClass, Class<?> rawType) {
+        for (JavaType anInterface : javaClass.getInterfaces()) {
+            if (anInterface.toErasure().isEquivalentTo(rawType)) {
+                return anInterface;
+            }
+        }
+        throw new AssertionError(String.format("Class %s has no interface of raw type %s", javaClass.getName(), rawType.getName()));
+    }
+
+    private interface SomeInterface {
+    }
+
+    @SuppressWarnings("unused")
+    private interface InterfaceWithSimpleTypeParameter<T> {
```

Comment:
Yeah, it's "simple" because it has only an unbound type parameter :joy: But you're right, the "simple" doesn't add much value, I'm gonna drop it :wink: